src/winbind.c

```diff
@@ -206,15 +206,76 @@ uint32_t winbind_cli_auth(char *user, char *domain,
     return ret;
 }
 
+/*  Worst-case SID in textual representation;
+    see https://stackoverflow.com/questions/1140528/
+        what-is-the-maximum-length-of-a-sid-in-sddl-format */
+static const size_t SID_MAX_STRING_LENGTH = 184 + 1;
+
+uint32_t put_sids_list_to_attributes(const struct wbcAuthUserInfo *wbc_info,
+                                     struct gssntlm_name_attribute **auth_attrs)
+{
+    static const char static_sids_urn[] = "urn:gssntlmssp:sids";
+    char *sids_buf;
+    char sid_str_buf[SID_MAX_STRING_LENGTH];
+    struct gssntlm_name_attribute *attrs = NULL;
+
+    size_t offset = 0;
+    size_t sids_list_len = 0;
+
+    /* Make a pass over the list of SIDs to find total length of textual data */
+    for (uint32_t i = 0; i < wbc_info->num_sids; i++) {
+        sids_list_len += wbcSidToStringBuf(&wbc_info->sids[i].sid, sid_str_buf,
+                                           sizeof(sid_str_buf));
+    }
+    /* +1 extra character per each SID for separator/EOL */
+    sids_list_len += wbc_info->num_sids;
+
+    /* Allocate buffers */
+    sids_buf = malloc(sids_list_len);
+    if (sids_buf == NULL) return ENOMEM;
+
+    /* 1 for returned attribute +1 for termiator entry */
+    attrs = malloc(sizeof(struct gssntlm_name_attribute) * 2);
```

Comment:
Use calloc() here and drop the code at the end that just add a bunch of zeros everywhere.

Reply:
Yep, good idea!